### PR TITLE
parser: implement Restore for ExistsSubqueryExpr

### PR DESCRIPTION
**Pull request by SwanSpouse**
Implement Restore and unit tests for ExistsSubqueryExpr

Issue: pingcap/tidb#8532

### CODE REVIEW

File: ast/expressions.go [hunk 1]
```diff
@@ -568,12 +568,29 @@ type ExistsSubqueryExpr struct {
 
 // Restore implements Node interface.
 func (n *ExistsSubqueryExpr) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if n.Not {
+		ctx.WriteKeyWord(" NOT EXISTS ")
+	} else {
+		ctx.WriteKeyWord(" EXISTS ")
```
[the remark this answers is not in the document]
Comment (SwanSpouse) [reply ast/expressions.go]: DONE PTAL

File: ast/expressions_test.go [hunk 2]
```diff
@@ -301,3 +301,15 @@ func (tc *testExpressionsSuite) TestMaxValueExprRestore(c *C) {
 	}
 	RunNodeRestoreTest(c, testCases, "alter table posts add partition ( partition p1 values less than %s)", extractNodeFunc)
 }
+
+func (tc *testExpressionsSuite) TestExistsSubqueryExprRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		// TODO: Test for subquery when it's implemented
+		//{"EXISTS (select 2)", "EXISTS (select 2)"},
```
Comment (leoppro) [inline ast/expressions_test.go]: i think we should merge this after `SelectStmt` implemented.

Comment (SwanSpouse) [reply ast/expressions_test.go]: OK

**Comment by leoppro** [comment]
`SelectStmt` has been implemented  #153 
@SwanSpouse

**Comment by SwanSpouse** [comment]
> `SelectStmt` has been implemented #153
> @SwanSpouse

PTAL

File: ast/expressions.go [hunk 3]
```diff
@@ -588,12 +588,25 @@ type ExistsSubqueryExpr struct {
...
+	if err := n.Sel.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occurred while restore ExistsSubqueryExpr.Sel")
+	}
+	return nil
 }
 
 // Format the ExprNode into a Writer.
 func (n *ExistsSubqueryExpr) Format(w io.Writer) {
-	panic("Not implemented")
+	if n.Not {
```
Comment (leoppro) [inline ast/expressions.go]: we don't need to implement the `Format` of `ExistsSubqueryExpr`

Comment (SwanSpouse) [reply ast/expressions.go]: OK

Comment (SwanSpouse) [reply ast/expressions.go]: PATL

**Comment by leoppro** [comment]
@kennytm PTAL

LGTM

**Comment by AndrewDi** [comment]
@SwanSpouse Please related test case in `parser_test.go`.

**Pull request merged**